File: AUGUST-2025/24-08-2025/p1493.java
class Solution {
    public int longestSubarray(int[] nums) {
        int ml=0;
        int pc=0;
        int cc=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==1){
                cc++;
            }else{
                ml=Math.max(ml,pc+cc);
                pc=cc;
                cc=0;
            }
            ml=Math.max(ml,pc+cc);
            if(ml==nums.length){
                ml--;
            }
        }
        return ml;
        

        
    }
}

File: AUGUST-2025/24-08-2025/p1493.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size();
        vector<int> zeroIdx;
        for(int i = 0; i < n; i++){
            if(nums[i] == 0){
                zeroIdx.push_back(i);
            }
        }

        if(zeroIdx.size() == 0)
        {
            return nums.size()-1;
        }
        int maxSize = 0;
        for(int i = 0; i < zeroIdx.size(); i++){
            int cntPre = 0, cntSuff = 0;
            for(int j = zeroIdx[i]-1; j >= 0; j--){
                if(nums[j] == 1){
                    cntPre++;
                }
                else{
                    break;
                }
            }

            for(int j = zeroIdx[i]+1; j < n; j++){
                if(nums[j] == 1){
                    cntSuff++;
                }
                else{
                    break;
                }
            }

            maxSize = max(maxSize, cntPre+cntSuff);
        }

        return maxSize;
    }
};